### DISCUSSION TITLE
Simple example of optimization

### DISCUSSION THREAD

**Discussion by Din-Y** [NanoComp/meep#2526] [General]
I am new at optimization and I am trying to apply adjoint solver via MEEP. Examples of adjoint solver from MEEP documentation seems to me so difficult and I cannot understand how to simplify your tutorial examples for my task. In all tutorial examples design region is divided into a 10х10 grid and each grid cell is optimized. I want to maximize the transmission around a silicon waveguide bend by changing parameters of dielectric mirror such as length, width, rotation angle and location of the coordinate center (please, see attached picture). Could you tell me please how I can implement the task to optimize only 4 – 5 parameters? Thank you!

![waveguide](https://github.com/NanoComp/meep/assets/132554077/e70cde6c-c761-4589-b65d-21e4ba6e4b34)

**Comment by stevengj** [comment]
Optimizing over a few geometric parameters like this are fundamentally different from the topology optimization considered in the manual.

If you only want to optimize over 4–5 parameters, just wrap the whole Meep simulation in a [derivative-free optimization](https://en.wikipedia.org/wiki/Derivative-free_optimization) algorithm (e.g. those from [NLopt](https://nlopt.readthedocs.io/en/latest/NLopt_Algorithms/) or other packages).
